esprima.js

```diff
@@ -4023,18 +4023,20 @@
         if (match('=')) {
             lex();
             init = isolateCoverGrammar(parseAssignmentExpression);
-        } else if (id.type !== Syntax.Identifier) {
-            expect('=');
+        } else {
+            if (!(options && options.inFor) && (id.type !== Syntax.Identifier)) {
+                expect('=');
+            }
         }
 
         return node.finishVariableDeclarator(id, init);
     }
 
-    function parseVariableDeclarationList() {
+    function parseVariableDeclarationList(options) {
         var list = [];
 
         do {
-            list.push(parseVariableDeclaration());
+            list.push(parseVariableDeclaration(options));
```

Comment:
Passing options through blindly bothers me a bit. Now one function's interface is dependent on the other's. This is hostile to change. I'd reconstruct the options object here.